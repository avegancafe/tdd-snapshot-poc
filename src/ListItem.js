import React, { Component } from "react";

export const ListItem = () => <div>Hello</div>;
