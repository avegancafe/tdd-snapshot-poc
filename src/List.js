import React, { Component } from "react";
import { ListItem } from "./ListItem";

export default class List extends Component {
  state = {
    showNegative: true
  };
  toggleNegative = () => {
    this.setState(previousState => {
      return {
        ...previousState,
        showNegative: !previousState.showNegative
      };
    });
  };
  get filteredArr() {
    return this.state.showNegative
      ? this.props.arr
      : this.props.arr.filter(el => el > 0);
  }
  render() {
    return (
      <div>
        {this.filteredArr.map((el, i) => (
          <ListItem key={i} value={el} />
        ))}
        <button onClick={this.toggleNegative} />
      </div>
    );
  }
}
