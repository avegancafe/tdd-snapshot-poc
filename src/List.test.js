import React from "react";
import ReactDOM from "react-dom";
import { shallow, configure } from "enzyme";
import Adaptor from "enzyme-adapter-react-16";
import { createSerializer } from "enzyme-to-json";

import List from "./List";
import { ListItem } from "./ListItem";

configure({ adapter: new Adaptor() });

expect.addSnapshotSerializer(createSerializer({ mode: "deep" }));

const arr = [1, 5, -3, 4, -2];

describe("List", () => {
  describe("when given a list of numbers", () => {
    let list;

    beforeEach(() => {
      list = shallow(<List arr={arr} />);
    });

    it("create corresponding list items", () => {
      expect(list.find(ListItem).length).toBe(arr.length);
      expect(list).toMatchSnapshot();
    });
  });

  describe("when clicking button", () => {
    let list;

    beforeEach(() => {
      list = shallow(<List arr={arr} />);
      let items = list.find(ListItem);
      expect(items.length).toBe(5);
      expect(list).toMatchSnapshot();

      list.find("button").simulate("click");

    });

    it("only shows positive numbers", () => {
      let items = list.find(ListItem);

      expect(items.length).toBe(3);

      items.map(item => {
        expect(item.prop("value") > 0).toBe(true);
      });
      expect(list).toMatchSnapshot();
    });
  });
});
