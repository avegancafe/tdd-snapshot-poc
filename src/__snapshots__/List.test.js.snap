// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List when clicking button only shows positive numbers 1`] = `
<div>
  <ListItem
    key="0"
    value={1}
  />
  <ListItem
    key="1"
    value={5}
  />
  <ListItem
    key="2"
    value={-3}
  />
  <ListItem
    key="3"
    value={4}
  />
  <ListItem
    key="4"
    value={-2}
  />
  <button
    onClick={[Function]}
  />
</div>
`;

exports[`List when clicking button only shows positive numbers 2`] = `
<div>
  <ListItem
    key="0"
    value={1}
  />
  <ListItem
    key="1"
    value={5}
  />
  <ListItem
    key="2"
    value={4}
  />
  <button
    onClick={[Function]}
  />
</div>
`;

exports[`List when given a list of numbers create corresponding list items 1`] = `
<div>
  <ListItem
    key="0"
    value={1}
  />
  <ListItem
    key="1"
    value={5}
  />
  <ListItem
    key="2"
    value={-3}
  />
  <ListItem
    key="3"
    value={4}
  />
  <ListItem
    key="4"
    value={-2}
  />
  <button
    onClick={[Function]}
  />
</div>
`;
